sw/device/freestanding/stdint.h

```diff
@@ -0,0 +1,139 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef SW_DEVICE_FREESTANDING_STDINT_H_
+#define SW_DEVICE_FREESTANDING_STDINT_H_
+
+/**
+ * This header implements the stdint.h standard header, as required by C11 S4.6.
+ * This header is specified in detail in S7.20 the same.
+ *
+ * Actual limit values below are defined as:
+ * - Unsigned MAXs are defined as the |0xff..ffU| all-ones representation.
+ * - Signed MINs are defined as |0x80..00| all-zeroes-but-the-last
+ *   representation.
+ * - Unsigned MAXs are defined as |0x7f..ff| all-ones-but-the-last
+ *   representation.
+ * These shorthands are correct for two's complement and the ones least likely
+ * to get a mistake in.
+ *
+ * It goes without saying that this file assumes the underlying representation
+ * is two's complement and little-endian.
+ *
+ * See https://github.com/riscv/riscv-elf-psabi-doc/blob/master/riscv-elf.md
+ * for the RISC-V specifics this file conforms to.
+ */
+
```

Comment:
You'll fine plenty of `__defines__` to help you implement this file, supported by both clang and gcc. See e.g. `riscv64-linux-gnu-gcc -march=rv32imc -mabi=ilp32 -dM -E - < /dev/null | sort`.

Reply:
I took a quick look at what was available, and I don't know how much *easier* it'd make things. There isn't anything available for the MIN versions of stuff, and a lot of stuff that would be useful for `limits.h` isn't available.

Defining all this stuff ourselves, explicitly, also at least gives us one source of truth for all our numeric definitions. Technically this shouldn't matter but I think it's good to be able to control i.e. unsigned int vs unsigned long.